src/dodal/devices/i19/shutter.py

```diff
@@ -1,61 +1,86 @@
-        self.shutter = HutchShutter(prefix=prefix, name=name)
-        bl_prefix = prefix.split("-")[0]
-        self.hutch_state = epics_signal_r(str, f"{bl_prefix}-OP-STAT-01:EHStatus.VALA")
-        if hutch == HutchState.INVALID:
-            raise HutchInvalidError(
-                "Cannot define experimental shutter for invalid hutch"
-            )
+        self.shutter_status = epics_signal_r(ShutterState, f"{prefix}STA")
         self.hutch_request = hutch
+        self.url = OPTICS_BLUEAPI_URL
         super().__init__(name)
 
+    async def read(self) -> dict[str, Reading]:
+        default_reading = await super().read()
+        status = await self.shutter_status.get_value()
+        return {
+            "shutter_status": Reading(value=status, timestamp=time.time()),
+            **default_reading,
+        }
+
     @AsyncStatus.wrap
     async def set(self, value: ShutterDemand):
-        hutch_in_use = await self.hutch_state.get_value()
-        LOGGER.info(f"Current hutch in use: {hutch_in_use}")
-        if hutch_in_use == HutchState.INVALID:
-            raise HutchInvalidError(
-                "The hutch state is invalid. Contact the beamline staff."
-            )
-        if hutch_in_use != self.hutch_request:
-            # NOTE Warn but don't fail
-            LOGGER.warning(
-                f"{self.hutch_request} is not the hutch in use. Shutter will not be operated."
-            )
-        else:
-            await self.shutter.set(value)
+        REQUEST_PARAMS = {
+            "name": "operate_shutter_plan",
+            "params": {
+                "experiment_hutch": self.hutch_request.value,
+                "access_device": "access_control",  # Device name in i19-blueapi
+                "shutter_demand": value,
+            },
+        }
+        async with ClientSession(base_url=self.url, raise_for_status=True) as session:
+            # First I need to submit the plan to the worker
+            async with session.post("/tasks", data=REQUEST_PARAMS) as response:
+                LOGGER.debug(
+                    f"Task submitted to the worker, response status: {response.status}"
+                )
+
+                try:
+                    data = await response.json()
+                    task_id = data["task_id"]
+                except Exception as e:
+                    LOGGER.error(
+                        f"Failed to get task_id from {self.url}/tasks POST. ({e})"
+                    )
+                    raise
+            # Then I can set the task as active and run asap
+            async with session.put(
+                "/worker/tasks", data={"task_id": task_id}
+            ) as response:
+                if not response.ok:
+                    LOGGER.warning(
+                        f"""Unable to operate the shutter.
+                        Session PUT responded with {response.status}: {response.reason}.
+                        """
+                    )
```

Comment:
Why is this just a warning and the last one an error? An issue in either would leave to us not being able to run the plan, is that an error or a warning?